dev/tools/merge-pr.sh

```diff
@@ -1,50 +1,126 @@
 #!/usr/bin/env bash
 
 set -e
+set -o pipefail
+
+API=https://api.github.com/repos/coq/coq
 
 # This script depends (at least) on git and jq.
 # It should be used like this: dev/tools/merge-pr.sh /PR number/
 
-#TODO: check arguments and show usage if relevant
+check_util() {
+if ! which $1 > /dev/null; then
+  echo error: this script requires the $1 command line utility
+  exit 1
+fi
+}
 
-PR=$1
+ask_confirmation() {
+  read -p "Continue anyway? [y/N] " -n 1 -r
+  echo
+  if [[ ! $REPLY =~ ^[Yy]$ ]]
+  then
+    exit 1
+  fi
+}
 
-CURRENT_LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-REMOTE=$(git config --get "branch.$CURRENT_LOCAL_BRANCH.remote")
-git fetch "$REMOTE" "refs/pull/$PR/head"
+RED="\033[31m"
+BLACK="\033[30m"
+GREEN="\033[32m"
+BLUE="\033[34m"
+YELLOW="\033[33m"
+info() {
+  echo -e "${GREEN}info:${BLACK}" $@ ${BLACK}
+}
+error() {
+  echo -e "${RED}error:${BLACK}" $@ ${BLACK}
+}
+warning() {
+  echo -e "${YELLOW}warning:${BLACK}" $@ ${BLACK}
+}
 
-API=https://api.github.com/repos/coq/coq
+check_util jq
+check_util curl
+check_util git
+check_util gpg
+
+# command line parsing
+
+if [ $# != 1 ]; then
+  echo usage: $0 PR-number
+  exit 1
+fi
+
+if [[ "$1" =~ ^[0-9][0-9]*$ ]]; then
+  PR=$1
+else
+  error $1 is not a number
+  exit 1
+fi
+
+# Fetching PR metadata
+
+TITLE=$(curl -s "$API/pulls/$PR" | jq -r '.title')
+info title for PR $PR is ${BLUE}$TITLE
 
 BASE_BRANCH=$(curl -s "$API/pulls/$PR" | jq -r '.base.label')
+info PR $PR targets branch ${BLUE}$BASE_BRANCH
+
+CURRENT_LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+info you are merging in ${BLUE}$CURRENT_LOCAL_BRANCH
+
+REMOTE=$(git config --get "branch.$CURRENT_LOCAL_BRANCH.remote")
+info remote for $CURRENT_LOCAL_BRANCH is ${BLUE}$REMOTE
 
+info fetching from $REMOTE the PR
+git remote update "$REMOTE"
+if ! git ls-remote "$REMOTE" | grep pull >/dev/null; then
+  error remote $REMOTE is not configured to fetch pull requests
+  error run: git config remote.$REMOTE.fetch +refs/pull/*/head:refs/remotes/$REMOTE/pr/*
+  exit 1
+fi
+git fetch "$REMOTE" "refs/pull/$PR/head"
 COMMIT=$(git rev-parse FETCH_HEAD)
-STATUS=$(curl -s "$API/commits/$COMMIT/status" | jq -r '.state')
+info commit for for PR $PR is ${BLUE}$COMMIT
```

Comment:
I don't understand what is going on but isn't there a typo here (repetition of `for`)

Reply:
typo